Library/OcDebugLogLib/OcLog.c

```diff
@@ -347,6 +455,55 @@ OcLogAddEntry  (
     }
   }
 
+  return Status;
+}
+
+EFI_STATUS
+EFIAPI
+OcLogAddEntry (
+  IN OC_LOG_PROTOCOL    *OcLog,
+  IN UINTN              ErrorLevel,
+  IN CONST CHAR8        *FormatString,
+  IN VA_LIST            Marker
+  )
+{
+  EFI_STATUS                  Status;
+
+  OC_LOG_PRIVATE_DATA         *Private;
+
+  UINTN                       Index;
+  BOOLEAN                     IsFiltered;
+  OC_LOG_FILTER_MODE          FilterMode;
+
+  ASSERT (OcLog != NULL);
+  ASSERT (FormatString != NULL);
+
+  Private = OC_LOG_PRIVATE_DATA_FROM_OC_LOG_THIS (OcLog);
+
+  if ((OcLog->Options & OC_LOG_ENABLE) == 0) {
+    //
+    // Silently ignore when disabled.
+    //
+    return EFI_SUCCESS;
+  }
+
+  //
+  // Filter log.
+  //
+  if (Private->FlexFilters != NULL) {
+    for (Index = 0; Index < Private->FlexFilters->Count; ++Index) {
```

Comment:
This outer loop looks wrong - you're already looping through these inside IsPrefixFiltered, no?

Reply:
https://github.com/acidanthera/OpenCorePkg/commit/bbaa7983d850dcd4c2c9813218babf52b3bd9445